examples/bri-1/base-example/test/shared.ts

```diff
@@ -138,11 +142,15 @@ export const shouldBehaveLikeAWorkgroupOrganization = function () {
 
     describe('workflow privacy', () => {
       let circuit;
-  
+
       describe('zkSNARK circuits', () => {
         describe('synchronization', () => {
           before(async () => {
-            circuit = await this.ctx.app.getBaselineCircuit();
+            // We need to Wait fo the Sync
+            while (getApp().getBaselineCircuit() === undefined) {
+              await sleep(10000);
```

Comment:
Since the last draft this is the part that i changed, i noticed that the tests we're not passing because it wasn't waiting for the Sync to happen. I added debugger notices and checked that usually waits at least 10s

Reply:
At this point in the test suite, we have already awaited the creation/delivery of the circuit. You're right in that the `await` is not necessary. But this just adds +10 seconds to the test suite without fixing any race conditions.

Reply:
Kyle, at my best understanding, comparing the base commit against this there is no change on the order of the actions on the test, but probably for the use of the function *tryTimes* it executes more early into this specific action, when the baseline circuit is not present in the app.
I added a debugger breakpoint and checked that usually with this changes the *baselineCircuit* is not yet populated on this specific commit, and sooner or later when this test evolves it should get to the same problem if it is not awaited to populate.

The exact times (10s) is a suggestion and can be changed to smaller times (5ms for example)